### PR TITLE
Store CRUD Cloud to Device

### PR DESCRIPTION
**Pull request by frankrowe**
## Status

**READY**
## JIRA Ticket

SPACON-150
## Description
- Listens for store create/update/delete actions from server
- Perform store operations
- Create Bridge observables for store and form lists
## Todos
- [ ] Tests
- [ ] Documentation
- [ ] License
## Steps to Test or Reproduce

``` sh
git fetch --all
git checkout <feature_branch> 
xctool -workspace SpatialConnect.xcworkspace -scheme SpatialConnect -sdk iphonesimulator -destination 'platform=iOS Simulator,name=iPhone 6 Plus' ONLY_ACTIVE_ARCH=NO test
```

@boundlessgeo/spatial-connect

### CODE REVIEW

File: SpatialConnect/SCBackendService.m [hunk 1]
```diff
@@ -88,18 +89,24 @@ - (void)setupSubscriptions {
...
-    default:
-      break;
+      case CONFIG_ADD_STORE: {
+        NSDictionary *payload = [json objectFromJSONString];
+        SCStoreConfig *config = [[SCStoreConfig alloc] initWithDictionary:payload];
+        [[[SpatialConnect sharedInstance] dataService] registerAndStartStoreByConfig:config];
+        break;
+      }
+      case CONFIG_UPDATE_STORE:
+        NSLog(@"Update Store");
```
Comment (frankrowe) [inline SpatialConnect/SCBackendService.m]: Add method to dataservice for update.
stop store, remove from list, register, start.

File: SpatialConnect/SCBackendService.m [hunk 2]
```diff
@@ -72,6 +72,7 @@ - (void)authListener {
     SCAuthStatus s = [n integerValue];
     if (s == SCAUTH_AUTHENTICATED) {
       [self registerAndFetchConfig];
+      [self fetchConfigAndListen];
```
Comment (tetriscode) [inline SpatialConnect/SCBackendService.m]: rename to listenForUpdates

File: SpatialConnect/SCBackendService.m [hunk 3]
```diff
@@ -88,18 +89,28 @@ - (void)setupSubscriptions {
...
-      break;
+      case CONFIG_ADD_STORE: {
+        NSDictionary *payload = [json objectFromJSONString];
+        SCStoreConfig *config = [[SCStoreConfig alloc] initWithDictionary:payload];
+        [[[SpatialConnect sharedInstance] dataService] registerAndStartStoreByConfig:config];
+        break;
+      }
+      case CONFIG_UPDATE_STORE: {
+        NSDictionary *payload = [json objectFromJSONString];
+        NSLog(@"UPDATE_STORE %@", payload);
```
Comment (tetriscode) [inline SpatialConnect/SCBackendService.m]: Drop this log

File: SpatialConnect/SCBackendService.m [hunk 4]
```diff
@@ -198,7 +201,15 @@ - (RACSignal *)publishReplyTo:(SCMessage *)msg onTopic:(NSString *)topic {
...
+    return [[multicast map:^SCMessage *(RACTuple *t) {
+      NSData *d = (NSData *)[t second];
+      NSError *err;
+      SCMessage *msg = [[SCMessage alloc] initWithData:d error:&err];
+      if (err) {
+        NSLog(@"%@", err.description);
+      }
+      return msg;
+    }] filter:^BOOL(SCMessage *m) {
       if (m.correlationId == msg.correlationId) {
```
Comment (tetriscode) [inline SpatialConnect/SCBackendService.m:210]: we should 
1. Subscribe to multicast
2. subscribe to channel
3. publish message
4. return multicast subscription

**Pull request merged**